internal/plan/plan.go

```diff
@@ -5,10 +5,14 @@ import (
 	"fmt"
 	"log/slog"
 	"os"
+	"sync"
+	"time"
 )
 
 type PlanHandler struct {
-	Path string `required:"True"`
+	logger *slog.Logger
+	mu     *sync.Mutex
```

Comment:
can you explain to me, why is this needed?
also, why is this a pointer to a mutex rather than the plain instance?

Reply:
The pointer is there to avoid an error from `govet` about copying lock by value: 
https://groups.google.com/g/golang-nuts/c/o-2N5wc8JcI

Reply:
Interesting